lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/MethodProcessor.java

```diff
@@ -611,29 +628,27 @@ private boolean generateWarningsCheck(
       return false;
     } else {
       out.println("    boolean anyWarnings = false;");
-      out.println("    ArrayRope<Warning> gatheredWarnings = new ArrayRope<>();");
+      out.println("    int maxWarnings = EnsoContext.get(bodyNode).getWarningsLimit();");
+      out.println("    EnsoHashMap gatheredWarnings = EnsoHashMap.empty();");
       for (var arg : argsToCheck) {
+        String argCode = arrayRead(argumentsArray, arg.getPosition());
         out.println(
             "    if ("
                 + arrayRead(argumentsArray, arg.getPosition())
-                + " instanceof WithWarnings) {");
+                + " instanceof WithWarnings withWarnings) {");
```

Comment:
Recently I was claiming to @GregoryTravis that `instanceof WithWarnings` isn't the right way to check for warnings. Is or isn't it the right way in this case? Is or isn't it the right way in [Greg's case](https://github.com/enso-org/enso/pull/10725#discussion_r1701382652)?

Reply:
In this case, checking via `warnsLibrary.hasWarnings(x)` is not possible. The reason is that later, we fetch the value out of `withWarnings` via `withWarnings.getValue()`, which retains the old warnings. If you checked for warnings with `warnsLibrary.hasWarnings(x)`, you would need to fetch the value via `warnsLibrary.removeWarnings(x)`, and that does not retain old warnings. I have already tried to implement it that way, but some tests in `Warning_Spec` on `Warning.get_all wrap_errors=True` were failing.

Reply:
> Is or isn't it the right way in [Greg's case](https://github.com/enso-org/enso/pull/10725#discussion_r1701382652)?

Generally, I would prefer the check to be done via `WarningsLibrary`. This case in `MethodProcessor` is a bit special, though.

Reply:
@GregoryTravis I have just [scheduled benchmarks](https://github.com/enso-org/enso/pull/10555#issuecomment-2271731842) for this PR. All other tests are already successful. If benchmarks are OK, I will merge it. Wait, please before I compare the benchmarks, I will ping you then.